src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1165,6 +1165,16 @@ object Desugar {
               _ <- write(in.New(target, zero)(src))
             } yield target
 
+          case PPredConstructor(base, args) => base match {
+            case PFPredBase(id) =>
+              val proxy = fpredicateProxyD(id)
+              val idT = typeD(info.typ(id), Addressability.rValue)(src).asInstanceOf[in.PredT]
```

Comment:
When you reverse the typing of predicaten, then here you will have to transforma a Functiontype to a predicate type.